consensus-engine/src/overlay/threshold.rs

```diff
@@ -0,0 +1,41 @@
+use fraction::{Fraction, GenericFraction, ToPrimitive};
+
+const LEADER_SUPER_MAJORITY_THRESHOLD_NUM: u64 = 2;
+const LEADER_SUPER_MAJORITY_THRESHOLD_DEN: u64 = 3;
+
+pub fn default_leader_super_majority_threshold() -> GenericFraction<u64> {
+    Fraction::new(
+        LEADER_SUPER_MAJORITY_THRESHOLD_NUM,
+        LEADER_SUPER_MAJORITY_THRESHOLD_DEN,
+    )
+}
+
+pub fn apply_threshold(size: usize, threshold: GenericFraction<u64>) -> usize {
+    // `threshold` is a tuple of (num, den) where `num/den` is the super majority threshold
+    (Fraction::from(size) * threshold)
+        .ceil()
```

Comment:
Previously, FlatOverlay was using `floor()`, and TreeOverlay was using `x * 2/3 + 1`. I thought it's better to have a common function like this and chose to use `ceil()` to guarantee that the threshold is never smaller than the super majority.